extensions-contrib/kubernetes-overlord-extensions/src/main/java/org/apache/druid/k8s/overlord/KubernetesPeonLifecycle.java

```diff
@@ -265,14 +268,32 @@ private TaskStatus getTaskStatus(long duration)
     return taskStatus.withDuration(duration);
   }
 
-  private void saveLogs()
+  protected void startWatchingLogs()
+  {
+    if (logWatch != null) {
+      log.debug("There is already a log watcher for %s", taskId.getOriginalTaskId());
+      return;
+    }
+    try {
+      Optional<LogWatch> maybeLogWatch = kubernetesClient.getPeonLogWatcher(taskId);
+      if (maybeLogWatch.isPresent()) {
```

Comment:
Do you need to handle multi-threaded access here? I see the `join` method is synchronized, but I'm not sure how this class is being accessed by many threads

Reply:
this method would only ever be called by one thread running join on that taskId. it's possible another thread (the one running shutdown commands) runs startWatchingLogs as well, but I don't think that would be an issue since logWatch would get set twice in succession which isn't too much of an issue